pyramid_openapi3/wrappers.py

```diff
@@ -1,76 +1,51 @@
 from pyramid.request import Request
 from pyramid.response import Response
+from urllib.parse import urljoin
 
 import typing as t
 
 
-class PyramidOpenAPIRequest(BaseOpenAPIRequest):
-    def __init__(self, request: Request) -> None:
-        self.request = request
-
-    @property
-    def host_url(self) -> str:
-        """Map openapi_core's host_url to pyramid's host_url."""
-        return self.request.host_url
-
-    @property
-    def path(self) -> str:
-        """Map openapi_core's path to pyramid's path."""
-        return self.request.path
-
-    @property
-    def method(self) -> str:
-        """Map openapi_core's methor to pyramid's method."""
-        return self.request.method.lower()
-
-    @property
-    def path_pattern(self) -> str:
-        """Map openapi_core's path_pattern to pyramid's path_pattern."""
-        if self.request.matched_route:
-            return self.request.matched_route.pattern
-        else:
-            return self.path
-
-    @property
-    def parameters(self) -> t.Dict:
-        """Map openapi_core's parameters to pyramid's request info."""
-        return {
-            "path": self.request.matchdict,
-            "query": self.request.GET,
-            "header": self.request.headers,
-            "cookie": self.request.cookies,
-        }
-
-    @property
-    def body(self) -> str:
-        """Map openapi_core's body to pyramid's body."""
-        return self.request.body
-
-    @property
-    def mimetype(self) -> str:
-        """Map openapi_core's mimetype to pyramid's content_type."""
-        return self.request.content_type
-
-
-class PyramidOpenAPIResponse(BaseOpenAPIResponse):
-    def __init__(self, response: Response):
-        self.response = response
-
-    @property
-    def data(self) -> bytes:
-        """Map openapi_core's data to pyramid's body."""
-        return self.response.body
-
-    @property
-    def status_code(self) -> int:
-        """Map openapi_core's status_code to pyramid's status_code."""
-        return self.response.status_code
-
-    @property
-    def mimetype(self) -> str:
-        """Map openapi_core's mimetype to pyramid's content_type."""
-        return self.response.content_type
+class PyramidOpenAPIRequest:
+    @classmethod
+    def create(
+        cls: t.Type["PyramidOpenAPIRequest"], request: Request
+    ) -> "OpenAPIRequest":
+        """Create OpenAPIRequest from Pyramid Request."""
+        method = request.method.lower()
+        path_pattern = (
+            request.matched_route.pattern if request.matched_route else request.path
+        )
+
+        parameters = RequestParameters(
+            path=request.matchdict,
+            query=request.GET,
+            header=request.headers,
+            cookie=request.cookies,
+        )
+        full_url_pattern = urljoin(request.host_url, path_pattern)
+
+        return OpenAPIRequest(
+            full_url_pattern=full_url_pattern,
```

Comment:
I think inheriting the `OpenAPIRequest` as a base class and returning `cls(...)` is a better design. So that it can be further extended/inherited.

Reply:
I was going with what openapi-core `contib` folder is doing for Django and Flask.

Reply:
I think so too. Either extend OpenAPIRequest class or go with a simple module level "factory method" instead of defining a single `@classmethod` on an otherwise unused class.

Reply:
I personally prefer the style you both suggested but I think in this case it is better to stick to the same pattern that openapi-core is using, for easier upgrades in the future.

Reply:
I went through the https://github.com/p1c2u/openapi-core codebase. Found these classes are (should be named) as `Factory` so that we can easily recognize them as factory classes instead of normal classes.